saleor/graphql/attribute/utils.py

```diff
@@ -417,6 +418,14 @@ def save(cls, instance: T_INSTANCE, cleaned_input: T_INPUT_MAP):
             associate_attribute_values_to_instance(
                 instance, attribute, *attribute_values
             )
+            if not attribute_values:
+                clean_assignment.append(attribute.pk)
+
+        # drop attribute assignment model when values are unassigned from instance
```

Comment:
Should we create a migration to drop old data?

Reply:
I added.